lib/include/tick/linear_model/model_hinge.h

```diff
@@ -21,19 +24,38 @@ class DLL_PUBLIC TModelHinge : public virtual TModelGeneralizedLinear<T> {
   using TModelGeneralizedLinear<T>::get_inner_prod;
   using TModelGeneralizedLinear<T>::get_class_name;
 
+ private:
+  TModelHinge() : TModelHinge<T>(nullptr, nullptr, false) {}
+
  public:
   TModelHinge(const std::shared_ptr<BaseArray2d<T> > features,
               const std::shared_ptr<SArray<T> > labels,
-              const bool fit_intercept, const int n_threads = 1);
+              const bool fit_intercept, const int n_threads = 1)
+      : TModelLabelsFeatures<T>(features, labels),
+        TModelGeneralizedLinear<T>(features, labels, fit_intercept, n_threads) {
+  }
 
   T loss_i(const ulong i, const Array<T> &coeffs) override;
 
   T grad_i_factor(const ulong i, const Array<T> &coeffs) override;
 
   template <class Archive>
   void serialize(Archive &ar) {
-    ar(cereal::make_nvp("ModelGeneralizedLinear",
-                        cereal::base_class<ModelGeneralizedLinear>(this)));
+    ar(cereal::make_nvp(
+        "ModelGeneralizedLinear",
+        typename cereal::base_class<TModelGeneralizedLinear<T> >(this)));
+  }
+
+  BoolStrReport compare(const TModelHinge<T> &that, std::stringstream &ss) {
```

Comment:
Can't this be generic and shared across models?

Reply:
Not as we have it no, the types don't exist on a superclass, not without implementing that templating thing I was telling you about.